src/AccessTokenCacheHandler.php

```diff
@@ -58,6 +58,8 @@ public function deleteItemByProvider(OAuth2Provider $provider, array $options):
 
     public function getCacheKey(OAuth2Provider $provider, array $options): string
     {
-        return static::CACHE_KEY_PREFIX . md5(print_r($provider, true) . serialize($options));
+        parse_str(parse_url($provider->getAuthorizationUrl(), PHP_URL_QUERY), $query);
+        return static::CACHE_KEY_PREFIX
+            . md5($provider->getBaseAuthorizationUrl() . ($query['client_id'] ?? '') . serialize($options));
```

Comment:
Shouldn't the field name "client_id" be injected instead of hardcoded here? I'm not sure this is an standard naming

Reply:
I didn't thought about it, but thankfully is standard :) https://tools.ietf.org/html/rfc6749#page-26

Reply:
👍